server/handlers.go

```diff
@@ -417,8 +422,21 @@ func (s *Server) notFoundHandler(w http.ResponseWriter, _ *http.Request) {
 	http.Error(w, http.StatusText(404), 404)
 }
 
+const newLineRegexp = "(\n|\r|\n\r)"
+
+var newLineRegexpCompiled = regexp.MustCompile(newLineRegexp)
+
 func sanitize(fileName string) string {
-	return path.Base(fileName)
+	t := transform.Chain(norm.NFD, runes.Remove(runes.In(unicode.Mn)), norm.NFC)
```

Comment:
not sure about `unicode.Mn`: we should allow diactrics, and as far as I can see this will drop.

In general I'm little lost in unicode composition/decomposition and compatibility/canonical

can you please explain what the transformation will look like for the following:
- `straße`
- `tëst`

Also we should apply `sanitize` everytime only in input, to remove extraneous character upon saving or processing

Especially is important to not introduce a breaking change when users will try to donwload files alread affected by the sanitization issue

Reply:
You right, I did too agressive normalization.
with  deleting unicode.Mn characters we get straße, without deleting straße
with  deleting unicode.Mn characters we get test, without deleting tëst
list of characters https://www.fileformat.info/info/unicode/category/Mn/list.htm

Reply:
> Also we should apply `sanitize` everytime only in input, to remove extraneous character upon saving or processing

As I can see you do this only on input (put, post, zip, gzip, etc). So previously uploaded file even with '\n\r' downloading after this commit.
Filenames can cantain a lot of unusual codes, even utf-16 or cesu8, and I afraid that it can cause problem in some cases, may be not with transfersh itself, but with http servers (frontends). I'm a little bit paranoic

Reply:
And I found way better than use regexp
```
	t := transform.Chain(
		norm.NFD,
		runes.Remove(runes.In(unicode.Cc)),
		runes.Remove(runes.In(unicode.Cf)),
		runes.Remove(runes.In(unicode.Co)),
		runes.Remove(runes.In(unicode.Other)),
		runes.Remove(runes.In(unicode.Zl)),
		norm.NFC)
```
https://pkg.go.dev/unicode#pkg-variables

How about this lists?

Reply:
that's good.

what' about adding `unicode.Cs` and `unicode.Zp`?

Reply:
> As I can see you do this only on input (put, post, zip, gzip, etc).

please, let me double check, when I looked briefly I was not to able to catch every usage :)

Reply:
> that's good.
> 
> what' about adding `unicode.Cs` and `unicode.Zp`?

I agree, it's a good offer

Reply:
great, fine :)

Reply:
> that's good.
> 
> what' about adding `unicode.Cs` and `unicode.Zp`?

Added this to transform chain
Double check usage of sanitize funciton. I see usage only in virustotal, clamav and postHandler, putHandler, zipHandler and tarGzHandler